src/fragments/lib/storage/flutter/upload.mdx

```diff
@@ -126,16 +182,271 @@ Future<void> uploadFile() async {
-        print('Fraction completed: ${progress.getFractionCompleted()}');
+        safePrint('Fraction completed: ${progress.getFractionCompleted()}');
       },
     );
-    print('Successfully uploaded file: ${result.key}');
+    safePrint('Successfully uploaded file: ${result.key}');
   } on StorageException catch (e) {
-    print('Error uploading file: $e');
+    safePrint('Error uploading file: $e');
+  }
+}
+```
+
+</Block>
+
+<Block name = "Developer Preview (Mobile, Web & Desktop)">
+
+```dart
+import 'package:file_picker/file_picker.dart';
+
+Future<void> uploadImage() async {
+  // Select a file from the device
+  final result = await FilePicker.platform.pickFiles(
+    type: FileType.custom,
+    withData: false,
+    // Ensure to get file stream for better performance
+    withReadStream: true,
+    allowedExtensions: ['jpg', 'png', 'gif'],
+  );
+
+  if (result == null) {
+    safePrint('No file selected');
+    return;
+  }
+
+  // Upload file with its filename as the key
+  final platformFile = result.files.single;
+  try {
+    final result = await Amplify.Storage.uploadFile(
+      localFile: AWSFile.fromStream(
+        platformFile.readStream!,
+        size: platformFile.size,
+      ),
+      key: platformFile.name,
+      onProgress: (progress) {
+        safePrint('Fraction completed: ${progress.fractionCompleted}');
+      },
+    ).result;
+    safePrint('Successfully uploaded file: ${result.uploadedItem.key}');
+  } on StorageException catch (e) {
+    safePrint('Error uploading file: $e');
+  }
+}
+```
+
+</Block>
+
+</BlockSwitcher>
+
+</Block>
+
+</BlockSwitcher>
+
+## Upload data (Developer Preview)
+
+To upload to S3 from a data object, specify the `key` and use `S3DataPayload` to create `data` object from various data formats.
+
+<BlockSwitcher>
+
+<Block name="String">
+
+```dart
+Future<void> uploadStringData({
+  required String dataString,
+  required String key,
+}) async {
+  try {
+    final result = await Amplify.Storage.uploadData(
+      data: S3DataPayload.string(
+        dataString,
+        contentType: 'text/plain',
+      ),
+      key: key,
+    ).result;
+
+    safePrint('Uploaded data: ${result.uploadedItem.key}');
+  } on S3Exception catch (e) {
+    safePrint(e.message);
+  }
+}
+```
+
+</Block>
+
+<Block name="JSON Object">
+
+```dart
+Future<void> uploadJsonObject({
+  required Map<String, Object> json,
+  required String key,
+}) async {
+  try {
+    final result = await Amplify.Storage.uploadData(
+      data: S3DataPayload.json(
+        json,
+        contentType: 'text/plain',
+      ),
+      key: key,
+    ).result;
+
+    safePrint('Uploaded data: ${result.uploadedItem.key}');
+  } on S3Exception catch (e) {
+    safePrint(e.message);
   }
 }
 ```
 
 </Block>
 
+<Block name="Data Url">
+
+```dart
+// `S3DataPayload.dataUrl()` throws exception on invalid data url string,
+// and extracts `contentType` value from the data url string.
+Future<void> uploadDataUrl({
```

Comment:
Can we explain what a "data URL" is here? I'm not sure most people would know, especially non-web developers.